### ISSUE TITLE
Allow explorer without collapse

### ISSUE THREAD

**Issue by pkra** [mathjax/MathJax-a11y#175]
Raised on the [user group](https://groups.google.com/forum/#!topic/mathjax-users/uS-O0aW7_O4).

It might be useful to allow the explorer to work without the collapsing extension.

**Comment by jfeferman** [comment]
Collapsible is an excellent tool for complex equations but may be overkill on content populated with simple equations (e.g. no reason to collapse "y = x + 1" while you may still want to use explorer for accessibility purposes). In some cases user experience could be enhanced by disabling the collapse feature while explorer is active.

**Comment by dpvc** [comment]
Just to be clear, collapsible does not make *everything* collapsible, only those subexpressions that are complicated enough.  So `y = x + 1` doesn't actually become collapsible.  Also, collapsible doesn't actually cause anything to *be* collapsed; it just makes it *possible* for them to be collapsed, if the user clicks on them.  So no expressions will be collapsed initially, and they only collapse if the reader specifically collapses them.  Finally, note that the explorer has keyboard shortcuts for collapsing and uncollapsing sub-expressions, and so even if it is collapsed and you want to explore it, you can.

Can you be more specific about how the user experience is enhanced by disabling the collapse feature during exploration?  It is not clear what you are concerned about specifically.

**Comment by jfeferman** [comment]
Thank you for your reply. Absolutely, collapsible works as you describe and is triggered by a user initiated action, either via mouse or keyboard. My example equation and simple expressions do not, as you point out, trigger collapsible but anything as simple as a quadratic equation does. Many equations in my project do trigger collapsible. 

My concern is that equation collapsing provides an extraneous feature to the core content and page interaction in my project. This could be viewed as a distraction and not add to the user experience, at the level of math we employ. Providing an option to disable this feature could be useful to developers, depending on the intended use and complexity of math.

**Comment by drlippman** [comment]
@dpvc, you said above that "no expression will be collapsed initially", but my experience was that while an equation like `f(x)=x^2` was not visually collapsed, the screenreader did read it as "collapsed equality."  Perhaps that's an unrelated issue, though.

**Comment by zorkow** [comment]
You are right, David. What is presented to a screen-reader in browse mode, exposed through an aria-label, is the most concise reading of a formula, i.e., the most collapsed version, regardless of whether a formula is visually collapsed or not.
The reason for this decision was that speaking every formula in its full glorious details adds considerably to the time it takes to screen read a document, not to mention the added cognitive load of trying to understand the content.

We are aware that the current solution is far from ideal. In particular, since our current heuristic on deciding what is collapsible or not is on based on a fairly simple complexity measure of sub-expression.

We had discussed a number of alternatives at the time, e.g., only summarising display equations, only summarising if an expression tree exceeds a certain depth.  Another simple idea would be to summarise only if the speech string exceeds a certain length. However, we never had the time to investigate that in more depth or define a more fine grained complexity measure for collapsing equations. 

But there is an open issue that one can not switch off collapsing entirely when the a11y extension is enabled. (I'll try to find that.) That would at least avoid any summaries, which could be useful for simple texts.

**Comment by zorkow** [comment]
Actually this is the open issue... I thought I was replying to issue #196 which is of course related.

**Comment by drlippman** [comment]
Thanks for the clarification, and I certainly understand the challenge finding time to explore alternatives :)  My application is a math homework system, and my concern was when a homework problem reads like "Consider the functions [collapsed equality] and [collapsed equality].  Evaluate [collapsed equality]".  I personally would have a hard time, while tabbing to each equation to read it,  keeping track where it fit back in the full question text.  But I suppose someone who regularly uses a screenreader would be much better at that than I am :)
